src/main/kotlin/uk/gov/justice/digital/hmpps/visitallocationapi/service/NomisSyncService.kt

```
@@ -107,29 +93,20 @@ class NomisSyncService(
       }
```

I notice that in syncPrisonerBalanceFromEventChange, calling getPrisonerDetails() is not the first thing you do, so the data from getBookingVisitBalances() could be stale.
Also if the prisoner is not found, isnt that an error? Might be better to throw an exception / return a 404 so the sync tries again. When messages are stuck on the queue the syscon team have alerts to flag this.

---

getBookingVisitBalances is an endpoint on the prison-api. This is an edge case where the syncPrisonerBalanceFromEventChange flow, needs to initially check if a balance exists on nomis, to then drive what we do next on dps.

if nomis returns a 404, we wipe our balance
if nomis returns a balance, we continue and sync

does that make sense?